modules/caddyhttp/celmatcher.go

```diff
@@ -520,47 +520,47 @@ func celMatcherStringMacroExpander(funcName string) parser.MacroExpander {
 //
 // The following function call is returned: <funcName>(request, arg)
 func celMatcherJSONMacroExpander(funcName string) parser.MacroExpander {
-	return func(eh parser.ExprHelper, target *exprpb.Expr, args []*exprpb.Expr) (*exprpb.Expr, *common.Error) {
+	return func(eh parser.ExprHelper, target ast.Expr, args []ast.Expr) (ast.Expr, *common.Error) {
 		if len(args) != 1 {
 			return nil, &common.Error{
 				Message: "matcher requires a map literal argument",
 			}
 		}
 		arg := args[0]
-		switch arg.GetExprKind().(type) {
-		case *exprpb.Expr_StructExpr:
-			structExpr := arg.GetStructExpr()
-			if structExpr.GetMessageName() != "" {
+		switch arg.Kind() {
+		case ast.MapKind:
+			structExpr := arg.AsStruct()
```

Comment:
Since this is a `MapKind` convert to a map using `AsMap()`, and then you won't have to do a type name check ... or you can add a `StructKind` case which immediately errors saying you want a map instead if you want to be helpful.